### ISSUE TITLE
Error in reproduce CamVid results

### ISSUE THREAD

**Issue by smajida** [GeorgeSeif/Semantic-Segmentation-Suite#75]
2. Bug reports
-----------------------
## Please specify the following information when submitting an issue:
- **What are your command line arguments?:**
`python main.py --num_epochs 200 --mode train --dataset CamVid --batch_size 1 --model FC-DenseNet103 --crop_height 352 --crop_width 480`
- **Have you written any custom code?**: yes
- **What have you done to try and solve this issue?:** nothing as the reason is not obvious
- **TensorFlow version?**: 1.8.0

### Describe the problem
I tried to reproduce the results mentioned by the author. 

However, after 6 epochs, loss explodes:
`[2018-06-10 20:56:42] Epoch = 6 Count = 20 Current_Loss = 257.05160522 Time = 8.38`
`...`
`[2018-06-10 21:18:04] Epoch = 13 Count = 20 Current_Loss = 224712.45312500 Time = 8.42`

I have modified the code mentioned below:
```

def load_image(path):
    image = cv2.cvtColor(cv2.imread(path,-1), cv2.COLOR_BGR2RGB)
    height, width, channels = image.shape
    resized_image =cv2.resize(image, (args.crop_width, args.crop_height))
    return resized_image
```

```
learning_rate = 0.001
rms_prop_decay = 0.995
opt = tf.train.RMSPropOptimizer(learning_rate =  learning_rate, decay = rms_prop_decay).minimize(loss)
```
****
### Source code / logs
```
Average validation accuracy for epoch # 0005 = 0.429434
Average per class validation accuracies for epoch # 0005:
Animal = 0.000000
Archway = 1.000000
Bicyclist = 0.600000
Bridge = 1.000000
Building = 0.427444
Car = 0.000000
CartLuggagePram = 0.500000
Child = 1.000000
Column_Pole = 0.000000
Fence = 0.800000
LaneMkgsDriv = 0.000000
LaneMkgsNonDriv = 1.000000
Misc_Text = 0.000000
MotorcycleScooter = 1.000000
OtherMoving = 0.000000
ParkingBlock = 0.600000
Pedestrian = 0.000601
Road = 0.560172
RoadShoulder = 1.000000
Sidewalk = 0.267866
SignSymbol = 0.400000
Sky = 0.796575
SUVPickupTruck = 0.600000
TrafficCone = 1.000000
TrafficLight = 0.400444
Train = 1.000000
Tree = 0.303346
Truck_Bus = 0.800011
Tunnel = 1.000000
VegetationMisc = 0.610922
Void = 0.000000
Wall = 0.500000
Validation precision =  0.3330594902286286
Validation recall =  0.42943359375000006
Validation F1 score =  0.36781106706617817
Validation IoU score =  0.09950381363969427
[2018-06-10 20:56:34] Remaining training time = 2 hours 14 minutes 2 seconds

[2018-06-10 20:56:42] Epoch = 6 Count = 20 Current_Loss = 257.05160522 Time = 8.38
[2018-06-10 20:56:51] Epoch = 6 Count = 40 Current_Loss = 463.60888672 Time = 8.45
[2018-06-10 20:56:59] Epoch = 6 Count = 60 Current_Loss = 960.87707520 Time = 8.42
[2018-06-10 20:57:08] Epoch = 6 Count = 80 Current_Loss = 525.40399170 Time = 8.58
```
![0001tp_006870_gt](https://user-images.githubusercontent.com/12903362/41205506-d8e17b38-6cf4-11e8-8f9a-4323ef74c8bb.png)
![0001tp_006870_pred](https://user-images.githubusercontent.com/12903362/41205507-d9072842-6cf4-11e8-8ad2-dea0e6a03754.png)

Class_dict.csv was not changed. Dataset was not changed.

**Comment by GeorgeSeif** [comment]
That simply looks like the loss is too high! Try decaying it or setting it lower initially.

As I mentioned in the README:

"These are some sample results for the CamVid dataset with 11 classes (previous research version)." The older CamVid research version had only 11 classes and the images were only of size 360x480, much easier to train on. So don't take the results at face value, try playing around with it :)

**Comment by smajida** [comment]
so you mean you did not use the dataset in the repository for training?

I used the learning rate of 0.001 with decaying factor of 0.995 and using RMSProp as mentioned in README. 

So I should not resize the images to 352x480?

**Comment by GeorgeSeif** [comment]
That's correct, I didn't use the dataset in the repository for training. The one I used for those results in the README is an older version of CamVid. This one is probably quite a bit more challenging to train on since the images are bigger and there are 32 classes.

You'll probably have to play around with the learning rate a bit.

In terms of image size, I think that size will be fine with the appropriate learning rate. For datasets with larger images, the usual way of doing things is to train on smaller sized images with a bigger batch size, then fine tune on the full size with batch size 1

**Comment by smajida** [comment]
got it. I just read about it as well. 32 classes were merged into 11 classes and images were resized to 360x480. First I will try to reproduce your reported results and then I will play around with the original size images. thanks.
